pkg/discord/main.go

```diff
@@ -8,24 +8,196 @@ import (
+						fmt.Println("error sending permissions message,", err)
+					}
+				}
+			}
+		case PermissionDM:
+			channel, err := s.Channel(m.ChannelID)
+			if err != nil {
+				return err
+			}
+			if channel.Type == discordgo.ChannelTypeDM || channel.Type == discordgo.ChannelTypeGroupDM {
+				command.Handler(cmdData)
+			} else {
+				if _, err = s.ChannelMessageSend(m.ChannelID, "This command is only accessible from a DM"); err != nil {
+					fmt.Println("error sending permissions message,", err)
+				}
+			}
+		case PermissionChannel:
+			channel, err := s.Channel(m.ChannelID)
+			if err != nil {
+				return err
+			}
+			if channel.Type == discordgo.ChannelTypeGuildText {
+				command.Handler(cmdData)
+			} else {
+				if _, err = s.ChannelMessageSend(m.ChannelID, "This command is only accessible from a server text channel"); err != nil {
+					fmt.Println("error sending permissions message,", err)
+				}
+			}
+		case PermissionAll:
+			command.Handler(cmdData)
+		}
+		return nil
+	} else {
+		if _, err := s.ChannelMessageSend(m.ChannelID, fmt.Sprintf("Unrecognized command, %v", key)); err != nil {
+			fmt.Println("error sending unrecognized command message,", err)
+			return err
+		}
+		return nil
+	}
+}
+
+// Register a command to the command handler.
+func (c *CommandHandler) RegisterCommand(cmd Command) {
+	c.Commands[cmd.Keyword] = cmd
+}
+
+// Create discord session and command handler
+func Create() (*discordgo.Session, error) {
+	dg, err := discordgo.New("Bot " + global.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	cmdHandler.Commands = make(map[string]Command)
+
+	onboardCommand := Command{
+		Keyword:    "onboard",
+		Handler:    onboard,
+		Permission: PermissionMembers,
+	}
+	cmdHandler.RegisterCommand(onboardCommand)
+	onboardAllCommand := Command{
+		Keyword:    "onboardall",
+		Handler:    onboardAll,
+		Permission: PermissionMembers,
+	}
+	cmdHandler.RegisterCommand(onboardAllCommand)
+	getAgendaCommand := Command{
+		Keyword:    "agenda",
+		Handler:    getAgenda,
+		Permission: PermissionAll,
+	}
+	cmdHandler.RegisterCommand(getAgendaCommand)
+	fetchFileCommand := Command{
+		Keyword:    "fetch",
+		Handler:    fetchFile,
+		Permission: PermissionMembers,
+	}
+	cmdHandler.RegisterCommand(fetchFileCommand)
+	joinCommand := Command{
+		Keyword:    "join",
+		Handler:    joinProject,
+		Permission: PermissionMembers,}
+	cmdHandler.RegisterCommand(joinCommand)
+	leaveCommand := Command{
+		Keyword:    "leave",
+		Handler:    leaveProject,
+		Permission: PermissionMembers,}
+	cmdHandler.RegisterCommand(leaveCommand)
+
+	return dg, nil
+}
+
+// When the bot connects to a server, record the number of uses on the onboarding invite, set role IDs.
 func ConnectToGuild(s *discordgo.Session, r *discordgo.Ready) {
 	for _, guild := range r.Guilds {
+		roles, err := s.GuildRoles(guild.ID)
+		if err != nil {
+			fmt.Println("error fetching guild roles,", err)
+		} else {
```

Comment:
want we want to stop here if we find an error instead of continuing on with the rest of the function ?

Reply:
@dviramontes No, because the next piece of the function(counting invites) is unrelated. This will also be removed in #60